servers/cromwell/jobs/controllers/utils/job_statuses.py

```diff
@@ -0,0 +1,65 @@
+from jobs.common import enum
+
+# The execution statuses of Cromwell are defined here:
+# https://github.com/broadinstitute/cromwell/blob/710c1931a6745a187ffd026f2cdafea2ffaaf2dc/core/src/main/scala/cromwell/core/ExecutionStatus.scala#L5
+# It provided a reasonable mapping in this file, but may need to be changed in the future on demand
+
+ApiStatus = enum(
+    SUBMITTED='Submitted',
+    ON_HOLD='OnHold',
+    RUNNING='Running',
+    ABORTING='Aborting',
+    ABORTED='Aborted',
+    FAILED='Failed',
+    SUCCEEDED='Succeeded')
+CromwellWorkflowStatus = enum(
+    SUBMITTED='Submitted',
+    ON_HOLD='On Hold',
+    RUNNING='Running',
+    ABORTING='Aborting',
+    ABORTED='Aborted',
+    FAILED='Failed',
+    SUCCEEDED='Succeeded')
+
+API_STATUS_MAP = {
+    ApiStatus.SUBMITTED: CromwellWorkflowStatus.SUBMITTED,
+    ApiStatus.ON_HOLD: CromwellWorkflowStatus.ON_HOLD,
+    ApiStatus.RUNNING: CromwellWorkflowStatus.RUNNING,
+    ApiStatus.ABORTING: CromwellWorkflowStatus.ABORTING,
+    ApiStatus.ABORTED: CromwellWorkflowStatus.ABORTED,
+    ApiStatus.FAILED: CromwellWorkflowStatus.FAILED,
+    ApiStatus.SUCCEEDED: CromwellWorkflowStatus.SUCCEEDED
+}
+
+CROMWELL_STATUS_MAP = {
+    CromwellWorkflowStatus.SUBMITTED: ApiStatus.SUBMITTED,
+    CromwellWorkflowStatus.ON_HOLD: ApiStatus.ON_HOLD,
+    CromwellWorkflowStatus.RUNNING: ApiStatus.RUNNING,
+    CromwellWorkflowStatus.ABORTING: ApiStatus.ABORTING,
+    CromwellWorkflowStatus.FAILED: ApiStatus.FAILED,
+    CromwellWorkflowStatus.SUCCEEDED: ApiStatus.SUCCEEDED
+}
+
+
+def api_workflow_status_to_cromwell(api_status):
+    """ Map an API workflow status to a Cromwell status.
+
+  :param str api_status: 'Submitted', 'On Hold', 'Running', 'Aborting', 'Failed', 'Succeeded'
+  :return: Cromwell status 'Submitted', 'OnHold', 'Aborting', 'Failed' or 'Succeeded'
+  """
+    if api_status not in API_STATUS_MAP:
+        raise ValueError(
+            'Unrecognized API execution status: {}'.format(api_status))
+    return API_STATUS_MAP[api_status]
+
+
+def cromwell_workflow_status_to_api(cromwell_status):
+    """ Map an API workflow status to a Cromwell status.
+
+  :param str cromwell_status: 'Submitted', 'OnHold', 'Aborting', 'Failed', 'Succeeded'
+  :return:  API status 'Submitted', 'On Hold', 'Running', 'Aborting', 'Failed' or 'Succeeded'
+  """
+    if cromwell_status not in CROMWELL_STATUS_MAP:
+        raise ValueError('Unrecognized Cromwell execution status: {}'.format(
```

Comment:
3 layers work together as the shim layer here: `controllers`(functions) -> `connexion` (API framework built on top of the Flask) -> `Gunicorn` Server.

So, if an exception in the controller has been thrown, e.g.:
```
cromwell_1    | ERROR:flask.app:Exception on /api/v1/jobs/query [POST]
cromwell_1    | Traceback (most recent call last):
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 2292, in wsgi_app
cromwell_1    |     response = self.full_dispatch_request()
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1815, in full_dispatch_request
cromwell_1    |     rv = self.handle_user_exception(e)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1718, in handle_user_exception
cromwell_1    |     reraise(exc_type, exc_value, tb)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1813, in full_dispatch_request
cromwell_1    |     rv = self.dispatch_request()
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1799, in dispatch_request
cromwell_1    |     return self.view_functions[rule.endpoint](**req.view_args)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/connexion/decorators/decorator.py", line 66, in wrapper
cromwell_1    |     response = function(request)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/connexion/decorators/validation.py", line 122, in wrapper
cromwell_1    |     response = function(request)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/connexion/decorators/validation.py", line 293, in wrapper
cromwell_1    |     return function(request)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/connexion/decorators/decorator.py", line 42, in wrapper
cromwell_1    |     response = function(request)
cromwell_1    |   File "/usr/local/lib/python2.7/dist-packages/connexion/decorators/parameter.py", line 195, in wrapper
cromwell_1    |     return function(**kwargs)
cromwell_1    |   File "/home/vmagent/app/jobs/controllers/utils/auth.py", line 14, in wrapper
cromwell_1    |     return fn(*args, **kwargs)
cromwell_1    |   File "/home/vmagent/app/jobs/controllers/jobs_controller.py", line 210, in query_jobs
cromwell_1    |     raise ValueError("I want an exception!")
cromwell_1    | ValueError: I want an exception!
```
The exception will be passed from controller to conexion and then to the low-level Flask framework, and Gunicorn will return a 500 error to the UI, which means the users will see `500: Internal Server Error` when a servre exception happens. 

I would say we need to use `try...except...finally...` as much as possible in controllers, to catch and mitigate those predictable exceptions at the 1st place. Furthermore, Flask/connexion supports adding exception handlers, which could be responsible for catching more exceptions. Finally, I think it would be better if UI could handle those 5xx errors more peacefully (like show a beautiful page or just snackBar) v.s. bare error page.

I'm not sure what is the equivalent in Python to the scala `scala.util.Try`, but try blocks + Exception handlers should be sufficient I think?